### PR TITLE
Upgrade nette

### PR DESCRIPTION
**Pull request by VBoss**
## Nette 3.0 compatibility
- Dropped PHP <7.1
- Support up to PHP 7.4 🐘 
- travis: drop kdyby/code-checker
- travis: run parallel-lint from vendor
- Exception split to own files 💥 
- setInject is deprecated, used addTag

---

👴 **Deprecated:**
- \Kdyby\RabbitMq\Exception::class => \Kdyby\RabbitMq\Exception\Exception::class
- \Kdyby\RabbitMq\InvalidArgumentException::class => \Kdyby\RabbitMq\Exception\InvalidArgumentException::class
- \Kdyby\RabbitMq\QueueNotFoundException::class => \Kdyby\RabbitMq\Exception\QueueNotFoundException::class
- \Kdyby\RabbitMq\TerminateException::class => \Kdyby\RabbitMq\Exception\TerminateException::class

---

⚠️ **BC Breaks:**
- Kdyby/RabbitMq/BaseConsumer::setCallback argument has type callable
- Kdyby/RabbitMq/BaseConsumer::setConsumerTag strict argument type string
- Kdyby/RabbitMq/BaseConsumer::getConsumerTag strict return type string
- Kdyby/RabbitMq/BaseConsumer::setQosOptions strict argument types
- Kdyby/RabbitMq/BaseConsumer::getIdleTimeout strict return type string
- Kdyby/RabbitMq/Connection::getConsumer strict argument type string
- Kdyby/RabbitMq/Connection::getProducer strict argument type string
- Kdyby/RabbitMq/Connection::getRpcClient strict argument type string
- Kdyby/RabbitMq/Connection::getRpcServer strict argument type string
- Kdyby/RabbitMq/Connection::doCreateChannel strict argument type string
- Kdyby/RabbitMq/Consumer::setMemoryLimit strict argument type
- **Kdyby/RabbitMq/Consumer::consume** strict argument type
- Kdyby/RabbitMq/Consumer::isRamAlmostOverloaded strict return type
- Kdyby/RabbitMq/DI/IConsumersProvider::getRabbitConsumers strict return type
- Kdyby/RabbitMq/DI/IProducersProvider::getRabbitProducers strict return type
- Kdyby/RabbitMq/DI/IRpcClientsProvider::getRabbitRpcClients strict return type
- Kdyby/RabbitMq/DI/IRpcServersProvider::getRabbitRpcServers strict return type
- Kdyby/RabbitMq/IConnection::getConsumer strict argument and return type
- Kdyby/RabbitMq/IConnection::getProducer strict argument and return type
- Kdyby/RabbitMq/IConnection::getRpcClient strict argument and return type
- Kdyby/RabbitMq/IConnection::getRpcServer strict argument and return type
- Kdyby/RabbitMq/IProducer::setRoutingKey strict argument and return type
- Kdyby/RabbitMq/IProducer::setContentType strict argument and return type
- Kdyby/RabbitMq/IProducer::setDeliveryMode strict argument and return type
- **Kdyby/RabbitMq/IProducer::publish** strict argument and return type
- Kdyby/RabbitMq/MultipleConsumer::processQueueMessage strict argument types
- Kdyby/RabbitMq/RpcClient::addRequest strict argument types
- Kdyby/RabbitMq/RpcServer::sendReply strict argument types

Thanks @VBoss & @jiripudil & @Jakub-Fajkus

### CODE REVIEW

**Comment by PavelJurasek** [comment]
https://github.com/Kdyby/Console/pull/80 might come in handy if it gets merged.

File: composer.json [hunk 1]
```diff
@@ -16,54 +16,56 @@
...
-		"nette/php-generator": "~2.4@dev",
-		"nette/reflection": "~2.4@dev",
-		"nette/robot-loader": "~2.4@dev",
-		"nette/safe-stream": "~2.3@dev",
-		"nette/security": "~2.4@dev",
-		"nette/tokenizer": "~2.3@dev",
-		"nette/utils": "~2.4@dev",
-		"latte/latte": "~2.4@dev",
-		"tracy/tracy": "~2.4@dev",
+        "kdyby/console": "~2.8",
```
Comment (fprochazka) [inline composer.json]: typo?

File: composer.json [hunk 2]
```diff
@@ -16,54 +16,56 @@
...
+	"autoload-dev": {
+		"psr-4": {
+			"KdybyTests\\RabbitMq\\": "tests/KdybyTests/RabbitMq"
+		}
 	},
+	"minimum-stability": "dev",
+	"repositories": [
+		{
+			"type": "git",
+			"url": "https://github.com/VBoss/Console.git"
```
Comment (fprochazka) [inline composer.json]: tohle by mělo jit pryč, ne?

**Comment by PavelJurasek** [comment]
```~2.8``` je bráno z repository ```vboss/console```, takže to lze odebrat, až bude nový release kdyby/console :)

**Comment by VBoss** [comment]
`2.8` je jako branch alias jen v composeru console. A to bylo včera zamergováno, ale tady mi to uteklo a při rebasu to mělo jít už pryč :)

**Comment by PavelJurasek** [comment]
Aaaa, rozumím! (TIL)

**Comment by foxycode** [comment]
@enumag Looks ok to me. Tests are passing. Will you merge it?

**Comment by enumag** [comment]
Sorry, I don't have time this week. Ping me on monday.

File: src/Kdyby/RabbitMq/Diagnostics/Panel.php [hunk 3]
```diff
@@ -102,43 +91,37 @@ public function getPanel()
...
 	}
 
-
-
-	/**
-	 * @param $message
-	 * @return object
-	 */
-	public function published($message)
+	public function published(string $message): void
```
Comment (MilanPala) [inline src/Kdyby/RabbitMq/Diagnostics/Panel.php]: Message se zde bere z `\Kdyby\RabbitMq\Channel::basic_publish` (https://github.com/Kdyby/RabbitMq/pull/72/files#diff-ecb3c979ad34881b9e2dd1f412a7c8e9R24), což generuje pole, nikoliv string.

Comment (MilanPala) [inline src/Kdyby/RabbitMq/Diagnostics/Panel.php]:
```suggestion
	public function published(array $message): void
```

**Comment by VBoss** [comment]
Pushnuto @MilanPala prosím zkus to jestli to funguje. Jinak Travis nějak zlobí a nechce nastartovat RabbitMq jako obvykle, ale dělá to jinak. Ještě to prověřím aby to procházelo

**Comment by RiKap** [comment]
@VBoss @enumag Ping :)

File: src/Kdyby/RabbitMq/Command/AnonConsumerCommand.php [hunk 4]
```diff
@@ -27,16 +22,14 @@ protected function configure()
...
-
-
-	protected function initialize(InputInterface $input, OutputInterface $output)
+	protected function initialize(InputInterface $input, OutputInterface $output): void
 	{
 		parent::initialize($input, $output);
 
 		if (!$this->consumer instanceof AnonymousConsumer) {
-			throw new InvalidArgumentException(
+			throw new \Kdyby\RabbitMq\Exception\InvalidArgumentException(
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/AnonConsumerCommand.php]: Add `use`.

File: src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php [hunk 5]
```diff
@@ -1,21 +1,15 @@
...
 use Symfony\Component\Console\Output\OutputInterface;
 
-
-
-/**
- * @author Alvaro Videla <videlalvaro@gmail.com>
- * @author Filip Procházka <filip@prochazka.su>
- */
-abstract class BaseConsumerCommand extends Command
+abstract class BaseConsumerCommand extends \Symfony\Component\Console\Command\Command
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php]: Add `use`

File: src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php [hunk 6]
```diff
@@ -25,7 +19,7 @@ abstract class BaseConsumerCommand extends Command
 	public $connection;
 
 	/**
-	 * @var Consumer|\Kdyby\RabbitMq\Consumer
+	 * @var \Kdyby\RabbitMq\Consumer
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php]: Add `use`

File: src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php [hunk 7]
```diff
@@ -34,113 +28,102 @@ abstract class BaseConsumerCommand extends Command
...
 			->addOption('without-signals', 'w', InputOption::VALUE_NONE, 'Disable catching of system signals');
 	}
 
-
-
 	/**
-	 * @param InputInterface $input An InputInterface instance
-	 * @param OutputInterface $output An OutputInterface instance
-	 *
+	 * @param \Symfony\Component\Console\Input\InputInterface $input An InputInterface instance
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php]:
```suggestion
```

File: src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php [hunk 8]
```diff
@@ -34,113 +28,102 @@ abstract class BaseConsumerCommand extends Command
...
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output An OutputInterface instance
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php]:
```suggestion
```

File: src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php [hunk 9]
```diff
@@ -34,113 +28,102 @@ abstract class BaseConsumerCommand extends Command
...
+		$this->amount = (int) $input->getOption('messages');
+		if ($this->amount < 0) {
+			throw new \InvalidArgumentException('The -m option should be null or greater than 0');
 		}
 
 		$this->consumer = $this->connection->getConsumer($input->getArgument('name'));
 
-		if (!is_null($input->getOption('memory-limit')) && ctype_digit((string) $input->getOption('memory-limit')) && $input->getOption('memory-limit') > 0) {
-			$this->consumer->setMemoryLimit($input->getOption('memory-limit'));
+		/** @var int|NULL $memoryLimit */
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php]:
```suggestion
		/** @var int|null $memoryLimit */
```

File: src/Kdyby/RabbitMq/Command/ConsumerCommand.php [hunk 10]
```diff
@@ -1,17 +1,13 @@
...
+declare(strict_types = 1);
 
+namespace Kdyby\RabbitMq\Command;
 
-/**
- * @author Alvaro Videla <videlalvaro@gmail.com>
- * @author Filip Procházka <filip@prochazka.su>
- */
-class ConsumerCommand extends BaseConsumerCommand
+class ConsumerCommand extends \Kdyby\RabbitMq\Command\BaseConsumerCommand
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/ConsumerCommand.php]:
```suggestion
class ConsumerCommand extends BaseConsumerCommand
```

File: src/Kdyby/RabbitMq/Command/PurgeConsumerCommand.php [hunk 11]
```diff
@@ -1,21 +1,15 @@
...
 use Symfony\Component\Console\Output\OutputInterface;
 
-
-
-/**
- * @author Alvaro Videla <videlalvaro@gmail.com>
- * @author Filip Procházka <filip@prochazka.su>
- */
-class PurgeConsumerCommand extends Command
+class PurgeConsumerCommand extends \Symfony\Component\Console\Command\Command
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/PurgeConsumerCommand.php]: Add `use`

File: src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php [hunk 12]
```diff
@@ -34,113 +28,102 @@ abstract class BaseConsumerCommand extends Command
...
-
-
 	/**
-	 * @param InputInterface $input An InputInterface instance
-	 * @param OutputInterface $output An OutputInterface instance
-	 *
-	 * @return integer 0 if everything went fine, or an error code
+	 * @param \Symfony\Component\Console\Input\InputInterface $input An InputInterface instance
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output An OutputInterface instance
 	 */
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/BaseConsumerCommand.php]: Remove docblock

File: src/Kdyby/RabbitMq/Command/RpcServerCommand.php [hunk 13]
```diff
@@ -1,20 +1,15 @@
...
 use Symfony\Component\Console\Output\OutputInterface;
 
-
-
-/**
- * @author Alvaro Videla <videlalvaro@gmail.com>
- * @author Filip Procházka <filip@prochazka.su>
- */
-class RpcServerCommand extends Command
+class RpcServerCommand extends \Symfony\Component\Console\Command\Command
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/RpcServerCommand.php]: Add use

File: src/Kdyby/RabbitMq/Command/RpcServerCommand.php [hunk 14]
```diff
@@ -23,36 +18,31 @@ class RpcServerCommand extends Command
...
 	/**
 	 * Executes the current command.
 	 *
-	 * @param InputInterface $input An InputInterface instance
-	 * @param OutputInterface $output An OutputInterface instance
-	 *
-	 * @return integer 0 if everything went fine, or an error code
-	 *
+	 * @param \Symfony\Component\Console\Input\InputInterface $input An InputInterface instance
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output An OutputInterface instance
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/RpcServerCommand.php]: Useless, remove.

File: src/Kdyby/RabbitMq/Command/SetupFabricCommand.php [hunk 15]
```diff
@@ -1,21 +1,15 @@
...
 use Symfony\Component\Console\Output\OutputInterface;
 
-
-
-/**
- * @author Alvaro Videla <videlalvaro@gmail.com>
- * @author Filip Procházka <filip@prochazka.su>
- */
-class SetupFabricCommand extends Command
+class SetupFabricCommand extends \Symfony\Component\Console\Command\Command
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/SetupFabricCommand.php]: add use

File: src/Kdyby/RabbitMq/Command/StdInProducerCommand.php [hunk 16]
```diff
@@ -1,20 +1,15 @@
...
 use Symfony\Component\Console\Output\OutputInterface;
 
-
-
-/**
- * @author Alvaro Videla <videlalvaro@gmail.com>
- * @author Filip Procházka <filip@prochazka.su>
- */
-class StdInProducerCommand extends Command
+class StdInProducerCommand extends \Symfony\Component\Console\Command\Command
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/StdInProducerCommand.php]: ...

File: src/Kdyby/RabbitMq/Command/StdInProducerCommand.php [hunk 17]
```diff
@@ -23,39 +18,33 @@ class StdInProducerCommand extends Command
...
-
 	/**
 	 * Executes the current command.
 	 *
-	 * @param InputInterface $input An InputInterface instance
-	 * @param OutputInterface $output An OutputInterface instance
-	 *
-	 * @return integer 0 if everything went fine, or an error code
+	 * @param \Symfony\Component\Console\Input\InputInterface $input An InputInterface instance
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output An OutputInterface instance
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Command/StdInProducerCommand.php]: rm

File: src/Kdyby/RabbitMq/Connection.php [hunk 18]
```diff
@@ -1,26 +1,21 @@
...
-/**
- * @author Filip Procházka <filip@prochazka.su>
- */
-class Connection extends PhpAmqpLib\Connection\AMQPLazyConnection implements IConnection
+class Connection extends \PhpAmqpLib\Connection\AMQPLazyConnection implements \Kdyby\RabbitMq\IConnection
 {
 
 	/**
-	 * @var Nette\DI\Container
+	 * @var \Nette\DI\Container
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Connection.php]: add use

File: src/Kdyby/RabbitMq/Connection.php [hunk 19]
```diff
@@ -1,26 +1,21 @@
...
 use Nette;
-use PhpAmqpLib;
-
 
-
-/**
- * @author Filip Procházka <filip@prochazka.su>
- */
-class Connection extends PhpAmqpLib\Connection\AMQPLazyConnection implements IConnection
+class Connection extends \PhpAmqpLib\Connection\AMQPLazyConnection implements \Kdyby\RabbitMq\IConnection
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Connection.php]: add use for both

File: src/Kdyby/RabbitMq/Connection.php [hunk 20]
```diff
@@ -1,26 +1,21 @@
...
 
 	/**
-	 * @var Nette\DI\Container
+	 * @var \Nette\DI\Container
 	 */
 	private $serviceLocator;
 
 	/**
-	 * @var Diagnostics\Panel
+	 * @var \Kdyby\RabbitMq\Diagnostics\Panel
```
Comment (enumag) [inline src/Kdyby/RabbitMq/Connection.php]: add use

**Comment by MilanPala** [comment]
Any progress to finish? We use this PR about two months without problem.

**Pull request merged**